File: src/main/java/tomekkup/helenos/types/qx/query/AbstractQuery.java
package tomekkup.helenos.types.qx.query;

/**
 * ********************************************************
 * Copyright: 2012 Tomek Kuprowski
 *
 * License: GPLv2: http://www.gnu.org/licences/gpl.html
 *
 * @author Tomek Kuprowski (tomekkuprowski at gmail dot com)
 * *******************************************************
 */
public abstract class AbstractQuery<K, N, V> implements Query<K, N, V> {

    private String keyspace;
    private String columnFamily;
    private Class<K> keyClass;
    private Class<N> nameClass;
    private Class<V> valueClass;
    private String consistencyLevel;
    
    public AbstractQuery() {
    }

    public AbstractQuery(Class<K> keyClass, Class<N> nameClass, String keyspace, String columnFamily) {
        this.keyClass = keyClass;
        this.nameClass = nameClass;
        this.keyspace = keyspace;
        this.columnFamily = columnFamily;
    }

    @Override
    public String getConsistencyLevel() {
        return consistencyLevel;
    }

    public void setConsistencyLevel(String consistencyLevel) {
        this.consistencyLevel = consistencyLevel;
    }

    @Override
    public String getKeyspace() {
        return keyspace;
    }

    public void setKeyspace(String keyspace) {
        this.keyspace = keyspace;
    }

    public String getColumnFamily() {
        return columnFamily;
    }

    public void setColumnFamily(String columnFamily) {
        this.columnFamily = columnFamily;
    }

    public void setKeyClass(Class<K> keyClass) {
        this.keyClass = keyClass;
    }

    public Class<K> getKeyClass() {
        return keyClass;
    }

    public void setNameClass(Class<N> nameClass) {
        this.nameClass = nameClass;
    }

    public Class<N> getNameClass() {
        return nameClass;
    }

    public void setValueClass(Class<V> valueClass) {
        this.valueClass = valueClass;
    }

    public Class<V> getValueClass() {
        return valueClass;
    }
}

File: src/main/frontend/source/class/helenos/model/AbstractQuery.js
/* ************************************************************************
Copyright:
  2012 Tomek Kuprowski
License:
  GPLv2: http://www.gnu.org/licences/gpl.html
Authors:
  Tomek Kuprowski (tomekkuprowski at gmail dot com)
 ************************************************************************ */
qx.Class.define('helenos.model.AbstractQuery', {
    extend : qx.core.Object,
    type : 'abstract',
    
    construct : function()
    {
        this.base(arguments);
    },

    properties :
    {
        consistencyLevel : {
            nullable : false, 
            init : 'ALL',
            check : 'String'
        },
        
        keyspace : {
            check : 'String',
            nullable : false,
            init : 'undefined'
        },
        
        columnFamily : {
            check : 'String',
            nullable : false,
            init : 'undefined'
        },
        
        keyClass : {
            check : 'String',
            nullable : false,
            init : 'undefined'
        },
        
        nameClass : {
            check : 'String',
            nullable : false,
            init : 'undefined'
        },
        
        valueClass : {
            check : 'String',
            nullable : false,
            init : 'undefined'
        }
    },
    
    members : {
        prepareQuery : function(cfDef, consistencyLevel) {
            this.setKeyClass(this._findParamClass(cfDef.keyValidationClass));
            if (cfDef.columnType == 'Standard') {
                this.setNameClass(this._findParamClass(cfDef.comparatorType.className));
            } else {
                this.setNameClass(this._findParamClass(cfDef.subComparatorType.className));
            }
            this.setValueClass(this._findParamClass(cfDef.defaultValidationClass));
            this.setKeyspace(cfDef.keyspaceName);
            this.setColumnFamily(cfDef.name);
            this.setConsistencyLevel(consistencyLevel);
        },
        
        _findParamClass : function(className) {
        switch (className)
            {
                case 'org.apache.cassandra.db.marshal.BytesType':
                case 'org.apache.cassandra.db.marshal.AsciiType':
                case 'org.apache.cassandra.db.marshal.UTF8Type':
                    return 'java.lang.String';
                    break;
                case 'org.apache.cassandra.db.marshal.LongType':
                case 'org.apache.cassandra.db.marshal.CounterColumnType':
                    return 'java.lang.Long';
                    break;
                case 'org.apache.cassandra.db.marshal.LexicalUUIDType':
                case 'org.apache.cassandra.db.marshal.TimeUUIDType':
                    return 'java.util.UUID';
                    break;
                default:
                    this.error('unknown class: ' + className);
                    return null;
                    break;
            }
        }
    }
});
